spec/controllers/drops_controller_spec.rb

```diff
@@ -99,12 +98,11 @@
     end
 
     context 'unsuccessfully updating a drop' do
-      let(:drop)   { build_stubbed :drop, :with_place, id: 2, sc_user_id: 654 }
-      let(:params) {{ id: 2, "place" => { "name" => "a sneaky hacker place", "location" => "sneaky hacker location" }}}
+      let(:drop)   { create :drop, :with_place, sc_user_id: 654 }
+      let(:params) {{ id: drop.id, "place" => { "name" => "a sneaky hacker place", "location" => "sneaky hacker location" }}}
 
       before do
         expect(controller).to receive(:current_user).and_return nil
-        expect(Drop).to receive(:find).with(drop.id).and_return drop
```

Comment:
In  app/controllers/drops_controller.rb 

we changed the #set_drop method and are no longer using the find method we were using there before, because we are now using  `page_not_found unless @drop = Drop.find_by(id: params[:id].to_i) `.  We didn't include this new ` find_by ` method in the test, because Carsten explained that the test shouldn't care about the methods by which the controller extracts data from the database, so the tests should test the outer layer more like black-box testing.